### ISSUE TITLE
Duplicate injections

### ISSUE THREAD

**Issue by cherrydev** [olov/ng-annotate#159]
As per issue #57 I am still seeing duplicate inject lines if you place an `// @ngInject` above a constructor that automatically gets discovered.  This is as of version 0.15.4.

**Comment by olov** [comment]
Hi! Please provide a minimal input example with ng-annotate output and I'll take a look. Thanks!

**Comment by olov** [comment]
(with input example I mean ES5 code that can be fed into ng-annotate. so most likely the output from your transpiler.)

**Comment by shahata** [comment]
I'm having this issue too:
```js
'use strict';

var x = (function () {
  /* @ngInject */
  function myFactory($http) {
    return $http;
  }
  return myFactory;
})();

angular.module('myApp').factory('x', x);
```

The above outputs this:
```js
var x = (function () {
  /* @ngInject */
  function myFactory($http) {
    return $http;
  }
  myFactory.$inject = ["$http"];
  myFactory.$inject = ["$http"];
  return myFactory;
})();

angular.module('myApp').factory('x', x);
```

**Comment by shahata** [comment]
btw, if you put the above output into ng-annotate you get an error:
error: conflicting inject arrays at line 8 and 9

I think ng-annotate can be more forgiving in case the dup inject arrays are identical...

**Comment by olov** [comment]
@shatata that's is a dupe of #135 and fixed in 79162568c53ebce941a49e4e2ab6464a45c39951, will be released in 1.0 quite soon hopefully. feel free to try master until then!

I'm not planning to relax ng-annotate complaining when there are multiple inject arrays.
